source/brailleDisplayDrivers/hidBrailleStandard.py

```diff
@@ -107,7 +107,8 @@ def __init__(self, port="auto"):
 			cellValueCaps = self._findCellValueCaps()
 			if cellValueCaps:
 				self._cellValueCaps = cellValueCaps
-				self.numCells = cellValueCaps.ReportCount
+				self._numberOfCellsValueCaps = self._findNumberOfCellsValueCaps()
+				self.numCells = self._maxNumberOfCells = cellValueCaps.ReportCount
```

Comment:
Can number of cells available be already less than the max number of cells when the screen reader connects to the device? If so, Is there a way for NvDA to fetch the initial value? Or can devices only decrease the number of cells once the screen reader is connected and receiving hid reports?

Reply:
Good point! There is a GetReport request in the USB spec for such scenarios, but software should generally use this only to test if a device was properly initialized.
In practice, the Braille display will send an initial input report with the correct number of cells very early on, so there may be a split second time frame in which the device expects less cells than NVDA assumes. Even if it doesn't, this isn't too much of a problem for two reasons:
1. The first keypress on the Braille display will contain the correct number of cells, and I made sure to set numCells before executing the input gesture.
2. The report count isn't affected by this parameter, so there will never be protocol errors due to a temporary mismatch.